src/lib/components/Post.svelte

```diff
@@ -99,18 +129,65 @@
         </Card.Footer>
     </div>
     <div class="flex flex-col items-center justify-between justify-self-end">
-        <Popover.Root>
-            <Popover.Trigger asChild let:builder>
-                <Button builders={[builder]} variant="ghost" size="icon" disabled={!post}>
-                    <div class:animate-pulse={!post}>
-                        <Share2 class="h-5 w-5" />
-                    </div>
-                </Button>
-            </Popover.Trigger>
-            <Popover.Content>
-                <h1>Not yet implemented!</h1>
-            </Popover.Content>
-        </Popover.Root>
+        <Dialog.Root bind:open>
+            <DropdownMenu.Root>
+                <DropdownMenu.Trigger asChild let:builder>
+                    <Button builders={[builder]} variant="ghost" size="icon" disabled={!post}>
+                        <div class:animate-pulse={!post}>
+                            <EllipsisVertical class="h-5 w-5" />
+                        </div>
+                    </Button>
+                </DropdownMenu.Trigger>
+                <DropdownMenu.Content class="w-56">
+                    <DropdownMenu.Group>
+                        <DropdownMenu.Item>Share</DropdownMenu.Item>
+                        <DropdownMenu.Separator />
+                        <Dialog.Trigger>Report</Dialog.Trigger>
+                        <Dialog.Content class="sm:max-w-[425px]">
+                            <Dialog.Header>
+                                <Dialog.Title>Report Form</Dialog.Title>
+                                <Dialog.Description>This action cannot be undone</Dialog.Description
+                                >
+                            </Dialog.Header>
+                            <div class="grid gap-2 py-2">
+                                <Select.Root bind:selected={reportData.reason} portal={null}>
+                                    <Select.Trigger class="w-[300px]">
+                                        <Select.Value
+                                            placeholder="What is the reason for the report?"
+                                        />
+                                    </Select.Trigger>
+                                    <Select.Content>
+                                        <Select.Group>
+                                            <Select.Label></Select.Label>
+                                            <Select.Item value="community"
+                                                >Post violates community guidlines.</Select.Item
+                                            >
+                                            <Select.Item value="site"
+                                                >Post violates site guidelines.</Select.Item
+                                            >
```

Comment:
The text for the items here should not have periods, also "guidelines" is misspelled in the `community` reason.

Reply:
When you fix that, can you also fix the formatting of these two items: 
```suggestion
                                            <Select.Item value="community">
                                            Post violates community guidlines.
                                            </Select.Item>
                                            <Select.Item value="site">
                                            Post violates site guidelines.
                                            </Select.Item>
```